### ISSUE TITLE
`anyio.run` positional parameters are not checked

### ISSUE THREAD

**Issue by injust** [astral-sh/ty#4433]
### Summary

```python
import anyio


async def main(a: int) -> None:
    print(a)


if __name__ == "__main__":
    anyio.run(main, "not an int")
```

basedpyright output:

```
/path/to/test.py:9:5 - error: Argument of type "tuple[Literal['not an int']]" cannot be assigned to parameter "args" of type "PosArgsT@run" in function "run"
  Type "*tuple[Literal['not an int']]" is not assignable to type "*tuple[int]"
    "*tuple[Literal['not an int']]" is not assignable to "*tuple[int]"
      Tuple entry 1 is incorrect type
        "Literal['not an int']" is not assignable to "int" (reportArgumentType)
```

But ty doesn't catch this.

### Version

0.0.76 via prek

**Comment by carljm** [comment]
Thanks! This is fixed by https://github.com/astral-sh/ruff/pull/27371
